cli/src/test-utils/test-runner.spec.ts

```diff
@@ -0,0 +1,209 @@
+    expect(scope.isDone()).toEqual(true);
+    expect(result).toEqual(true);
+  });
+
+  test("multiple provider states", async () => {
+    const contract = parseAndCleanse(
+      "./cli/src/test-utils/test-runner-examples/multiple-provider-states.ts"
+    );
+
+    const scope = nock(baseUrl)
+      .get("/companies/abc/users/def")
+      .reply(200, {
+        firstName: "John",
+        lastName: "Snow",
+        email: "johnsnow@spot.com"
+      })
+      .post("/state", { name: "a company exists", params: { id: "abc" } })
+      .query({ action: "setup" })
+      .reply(200)
+      .post("/state", {
+        name: "a user exists",
+        params: { id: "def", companyId: "abc" }
+      })
+      .query({ action: "setup" })
+      .reply(200)
+      .post("/state")
+      .query({ action: "teardown" })
+      .reply(200);
+
+    const result = await runTest(contract, stateUrl, baseUrl);
+    expect(scope.isDone()).toEqual(true);
+    expect(result).toEqual(true);
+  });
+
+  test("provider state setup fail", async () => {
+    const contract = parseAndCleanse(
+      "./cli/src/test-utils/test-runner-examples/single-provider-state.ts"
+    );
+
+    const scope = nock(baseUrl)
+      .post("/state", { name: "a company exists", params: { id: "abc" } })
+      .query({ action: "setup" })
+      .reply(400)
+      .post("/state")
+      .query({ action: "teardown" })
+      .reply(200);
+
+    const result = await runTest(contract, stateUrl, baseUrl);
+    expect(scope.isDone()).toEqual(true);
+    expect(result).toEqual(false);
+  });
+
+  test("provider state teardown fail", async () => {
+    const contract = parseAndCleanse(
+      "./cli/src/test-utils/test-runner-examples/single-provider-state.ts"
+    );
+
+    const scope = nock(baseUrl)
+      .get("/companies/abc")
+      .reply(200, {
+        name: "My Company",
+        employeeCount: 15
+      })
+      .post("/state", { name: "a company exists", params: { id: "abc" } })
+      .query({ action: "setup" })
+      .reply(200)
+      .post("/state")
+      .query({ action: "teardown" })
+      .reply(400);
+
+    const result = await runTest(contract, stateUrl, baseUrl);
+    expect(scope.isDone()).toEqual(true);
+    expect(result).toEqual(false);
+  });
+
+  test("response status mismatch", async () => {
+    const contract = parseAndCleanse(
+      "./cli/src/test-utils/test-runner-examples/single-provider-state.ts"
+    );
+
+    const scope = nock(baseUrl)
+      .get("/companies/abc")
+      .reply(204, {
+        name: "My Company",
+        employeeCount: 15
+      })
+      .post("/state", { name: "a company exists", params: { id: "abc" } })
+      .query({ action: "setup" })
+      .reply(200)
+      .post("/state")
+      .query({ action: "teardown" })
+      .reply(200);
+
+    const result = await runTest(contract, stateUrl, baseUrl);
+    expect(scope.isDone()).toEqual(true);
+    expect(result).toEqual(false);
```

Comment:
I think it would be valuable to also test for specific error messages. Could be done in a separate PR though :)